### PR TITLE
Home w26.2

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

**Comment by OleksiiTromsa** [comment]
Seems to me you forgot to imlement page with order details. 
"Создать две страницы с заказами (с возможностью перейти на “детали заказа”)"

File: src/main/java/com/internet/shop/controller/DeleteProductFormShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ProductService;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductFormShoppingCartController extends HttpServlet {
```
Comment (OleksiiTromsa) [inline src/main/java/com/internet/shop/controller/DeleteProductFormShoppingCartController.java]:
```suggestion
public class DeleteProductFromShoppingCartController extends HttpServlet {
```
Form -> From

File: src/main/java/com/internet/shop/controller/GetAdminOrdersController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Order;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAdminOrdersController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/GetAdminOrdersController.java]:
```suggestion
public class GetOrdersController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controller/GetAllAdminProductsController.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllAdminProductsController extends HttpServlet {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/GetAllAdminProductsController.java]:
```suggestion
public class ManageProductsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controller/DeleteProductController.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        Long productId = Long.valueOf(id);
+        productService.deleteById(productId);
+        resp.sendRedirect(req.getContextPath() + "/products/all/admin");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/DeleteProductController.java]: we should do the same after we added a new product to the storage in AddProductController: `resp.sendRedirect(req.getContextPath() + "/products/all/admin");`

File: src/main/webapp/WEB-INF/views/order/orderUser.jsp [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        <th>Products</th>
+    </tr>
+    <c:forEach var="order" items="${orders}">
+        <tr>
+            <td>
+                <c:out value="${order.id}"/>
+            </td>
+            <td>
+                <c:out value="${order.products}"/>
+            </td>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/orderUser.jsp]: here we should have the `Details` button instead that will show us a separate jsp with details
let's create that jsp and OrderDetailsController

File: src/main/webapp/WEB-INF/views/order/ordersForAdmin.jsp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        <tr>
+            <td>
+                <c:out value="${order.id}"/>
+            </td>
+            <td>
+                <c:out value="${order.userId}"/>
+            </td>
+            <td>
+                <c:out value="${order.products}"/>
+            </td>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/ordersForAdmin.jsp]: same here

File: src/main/webapp/web.xml [hunk 7]
```diff
@@ -89,4 +89,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>createOrder</servlet-name>
+        <servlet-class>com.internet.shop.controller.CompleteOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>createOrder</servlet-name>
+        <url-pattern>/order/create</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml:110]:
```suggestion
        <url-pattern>/order/complete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -89,4 +89,74 @@
...
     </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteProductFromShoppingCart</servlet-name>
+        <servlet-class>com.internet.shop.controller.DeleteProductFormShoppingCartController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteProductFromShoppingCart</servlet-name>
+        <url-pattern>/shopping-carts/products/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/shopping-cart/products/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 9]
```diff
@@ -89,4 +89,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allOrders</servlet-name>
+        <servlet-class>com.internet.shop.controller.GetUserOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allOrders</servlet-name>
+        <url-pattern>/users/order</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/user/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 10]
```diff
@@ -89,4 +89,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>adminOrder</servlet-name>
+        <servlet-class>com.internet.shop.controller.GetAdminOrdersController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>adminOrder</servlet-name>
+        <url-pattern>/order/admin</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders</url-pattern>
```

File: src/main/webapp/web.xml [hunk 11]
```diff
@@ -89,4 +89,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>deleteOrder</servlet-name>
+        <servlet-class>com.internet.shop.controller.DeleteOrderController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>deleteOrder</servlet-name>
+        <url-pattern>/order/delete</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/orders/delete</url-pattern>
```

File: src/main/webapp/web.xml [hunk 12]
```diff
@@ -89,4 +89,74 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>allAdminProducts</servlet-name>
+        <servlet-class>com.internet.shop.controller.GetAllAdminProductsController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>allAdminProducts</servlet-name>
+        <url-pattern>/products/all/admin</url-pattern>
```
Comment (KseniiaMakarova) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/products/manage</url-pattern>
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controller/DeleteProductFormShoppingCartController.java (hunk 1 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/DeleteProductFormShoppingCartController.java]: Done

File: src/main/java/com/internet/shop/controller/GetAdminOrdersController.java (hunk 2 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/GetAdminOrdersController.java]: Done

File: src/main/java/com/internet/shop/controller/GetAllAdminProductsController.java (hunk 3 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/GetAllAdminProductsController.java]: Done

File: src/main/java/com/internet/shop/controller/DeleteProductController.java (hunk 4 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/controller/DeleteProductController.java]: Done

File: src/main/webapp/web.xml (hunk 8 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 9 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 10 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 11 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/web.xml (hunk 12 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/web.xml]: Done

File: src/main/webapp/WEB-INF/views/order/ordersForAdmin.jsp (hunk 6 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/WEB-INF/views/order/ordersForAdmin.jsp]: Done

File: src/main/webapp/WEB-INF/views/order/orderUser.jsp (hunk 5 above)
Comment (KostiantynVorobiov) [reply src/main/webapp/WEB-INF/views/order/orderUser.jsp]: Done

File: src/main/java/com/internet/shop/controller/AddProductController.java [hunk 13]
```diff
@@ -27,6 +27,6 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
         Double price = Double.valueOf(req.getParameter("price"));
         Product product = new Product(name, price);
         productService.create(product);
-        resp.sendRedirect(req.getContextPath() + "/");
+        resp.sendRedirect(req.getContextPath() + "/products/all/admin");
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/AddProductController.java:30]:
```suggestion
        resp.sendRedirect(req.getContextPath() + "/products/manage");
```

File: src/main/java/com/internet/shop/controller/ShowDetailsOrderController.java [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/orders/details")
+public class ShowDetailsOrderController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final OrderService orderService =
+            (OrderService) injector.getInstance(OrderService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/ShowDetailsOrderController.java:21]: do you use this?

File: src/main/java/com/internet/shop/controller/ShowDetailsOrderController.java [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        Long orderId = Long.valueOf(id);
+        List<Product> products = orderService.getById(orderId).getProducts();
+        req.setAttribute("products", products);
+        req.setAttribute("orderId", orderId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/ShowDetailsOrderController.java:30]: we can set only one attribute here - `order`
we can get id and products in JSP

File: src/main/webapp/WEB-INF/views/order/orderUser.jsp [hunk 16]
```diff
@@ -0,0 +1,28 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>User's order</title>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/orderUser.jsp:5]:
```suggestion
    <title>User's orders</title>
```

File: src/main/webapp/WEB-INF/views/order/ordersForAdmin.jsp [hunk 17]
```diff
@@ -0,0 +1,36 @@
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<html>
+<head>
+    <title>Admin's Order</title>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/order/ordersForAdmin.jsp:5]:
```suggestion
    <title>All orders</title>
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**